### ISSUE TITLE
step_downsample clarification needed.

### ISSUE THREAD

**Issue by ronencozen** [tidymodels/recipes#320]
Hi,
I would like to validate my understanding.
if step_downsample is part of a recipe then the recipe could not be used in caret::train,
but it should be juiced first to materialized the preprocessed training dataset.
Unless skip is set to FALSE, then you can't use the recipe to preprocess the test set.

**Comment by topepo** [comment]
You could add a down-sampling step to a recipe used in `caret`. 

`caret` passes the modeling data set to the recipe and the recipe is prepped on that. The recipe is applied to the modeling set and those data are given to the model function. This is exactly what any of the options in `preProc` would do. 

You can place `step_downsample` anywhere in the recipe so you can decide if it should happen first or last, etc. 

> Unless skip is set to FALSE, then you can't use the recipe to preprocess the test set.

You need to be more specific. 

 - If you mean the holdout set used within `train`, then yes but you should never down-sample those data. 

 - If you mean new data points being given to `predict()`, you should also never down-sample those either. 

The one caveat right now is a [bug](https://github.com/topepo/caret/issues/1030) in `train` that I'll fix by the end of the week. If you use down-sampling with `train`, it might cause an error.

**Comment by ronencozen** [comment]
At first, I tried to run caret with the recipe that includes the downsampling step and ran into the bug that you indicated. I opened the documentation for downsampling and there you advise to retain and juice the recipe.

>  It is advisable to use prep(recipe, retain = TRUE) when preparing the recipe; in this way juice() can be used to obtain the down-sampled version of the data.

this is how I came to the conclusion above.

**Comment by topepo** [comment]
>  I opened the documentation for downsampling and there you advise to retain and juice the recipe.

Which documentation?

**Comment by ronencozen** [comment]
Package ‘recipes’ November 19, 2018 documentation. Page #50 second sentence under details.
In addition to your comment above, indeed a recipe that includes downsampling step should not be materialized and be used for modeling since then the model is prone to overfitting.
